### PR TITLE
Class encode labels column

### PR DESCRIPTION
**Pull request by ghomasHudson**
I think this is the fix needed for #154. Huggingface datasets now has a `class_encode_column` function which converts a column to `ClassLabel` which allows us to find the `num_classes`.

### CODE REVIEW

File: lightning_transformers/task/nlp/text_classification/data.py [hunk 1]
```diff
@@ -37,6 +37,9 @@ def process_data(self, dataset: Dataset, stage: Optional[str] = None) -> Dataset
         cols_to_keep = [
             x for x in ["input_ids", "attention_mask", "token_type_ids", "labels"] if x in dataset["train"].features
         ]
+        for split in dataset.keys():
```
Comment (Borda) [inline lightning_transformers/task/nlp/text_classification/data.py:40]:
```suggestion
        for split in dataset:
```

**Review by Borda — APPROVED** [review]

**Comment by SeanNaren** [comment]
Interesting I didn't know about the prepare for task function! Thanks for highlighting this :)

**Comment by Borda** [comment]
> Interesting I didn't know about the prepare for task function! Thanks for highlighting this :)

shall we merge it then?
cc: @carmocca

**Review by carmocca — APPROVED** [review]
